docs/usage/validators.md

```diff
@@ -1,375 +1,244 @@
-    def check_sum(cls, v):
-        if sum(v) > 42:
-            raise ValueError('sum of numbers greater than 42')
-        return v
 
-
-print(DemoModel(square_numbers=[1, 4, 9]))
-#> square_numbers=[1, 4, 9] cube_numbers=[]
-print(DemoModel(square_numbers='1|4|16'))
-#> square_numbers=[1, 4, 16] cube_numbers=[]
-print(DemoModel(square_numbers=[16], cube_numbers=[8, 27]))
-#> square_numbers=[16] cube_numbers=[8, 27]
+print(DemoModel(number=[2, 8]))
+#> number=[2, 8]
+print(DemoModel.model_validate_json(json.dumps({'number': [' 2 ', '8']})))
+#> number=[2, 8]
 try:
-    DemoModel(square_numbers=[1, 4, 2])
+    DemoModel(number=['2'])
 except ValidationError as e:
     print(e)
     """
     1 validation error for DemoModel
-    square_numbers.2
-      Assertion failed, 2 is not a square number
-    assert ((2 ** 0.5) % 1) == 0 [type=assertion_error, input_value=2, input_type=int]
-    """
-
-try:
-    DemoModel(cube_numbers=[27, 27])
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for DemoModel
-    cube_numbers
-      Value error, sum of numbers greater than 42 [type=value_error, input_value=[27, 27], input_type=list]
+    number.0
+      Assertion failed, In Python mode the input must be an int!
+    assert False
+     +  where False = isinstance('2', int) [type=assertion_error, input_value='2', input_type=str]
     """
 ```
 
-!!! note "Validation order"
-    Validation order matters. Within a given type, validation goes from right to left and back. That is, it goes
-    from right to left running all "before" validators (or calling into "wrap" validators), then left to right back out calling all "after" validators.
+The same "modes" apply to `@field_validator`, which is discussed in the next section.
 
-    In the following example, `func2` will be called before `func1`.
+### Ordering of validators within `Annotated`
 
-    ```py test="skip" lint="skip" upgrade="skip"
-    MyVal = Annotated[int, AfterValidator(func1), BeforeValidator(func2)]
-    ```
+Order of validation metadata within `Annotated` matters.
+Validation goes from right to left and back.
+That is, it goes from right to left running all "before" validators (or calling into "wrap" validators), then left to right back out calling all "after" validators.
 
-A few more things to note:
+In the following example, `func2` will be called before `func1`.
 
-* A single validator can be applied to multiple fields by passing it multiple field names.
-* A single validator can also be called on *all* fields by passing the special value `'*'`.
-* The keyword argument `mode='before'` will cause the validator to be called prior to other validation.
-* Using validator annotations inside of `Annotated` allows applying validators to items of collections.
+```py test="skip" lint="skip" upgrade="skip"
+MyVal = Annotated[int, AfterValidator(func1), BeforeValidator(func2)]
+```
 
-## Special Types
+## Field validators
```

Comment:
I feel like it might make more sense to start with this section than the annotated validators, since 1, this is closer to existing usage, and 2, it doesn't introduce the concept of using `Annotated`. I feel like it introduces just the concept of a validator function, which then motivates the idea of using `Annotated` validators as a way to make them reusable by way of a reused type annotation (rather than needing to set up the validator on every model you want to use it with).

I'm fine if you disagree, I just think this topic is quicker to understand and lowers the barrier to understand the annotated ones.